### ISSUE TITLE
How to play multiplayer in MameAppleTV

### ISSUE THREAD

**Issue by johnstewart** [kevsmithpublic/MameAppleTV#12]
Does anyone know if it's possible to get 2 (or more) controllers working simultaneously?

For those games supporting multiplayer, but with one controller, is there a way to hit player-2 start with the MFi remote?

**Comment by ian-a-anderson** [comment]
If you hit the start button while holding "Up" on the d-pad/joystick it will send the player-2 start command. This works, but I've had issues trying to get two MFI controllers (both SteelSeries Nimbus) to register as separate controllers. When I open MAME on the AppleTV, both controllers show registered as Player 1 (the first light of the four LEDs on the control pad) and only one of the controllers works. What is bizarre is that if I start a 2-player game using the method above, that single controller controls both players simultaneously. Not sure if that is a bug specific to the Nimbus or not, although it would be interesting to know what your experience is.

**Comment by romadu** [comment]
I had the same issue with Nimbus. I hadn't realised the trick to recognise
the second controller.
Provenance has a nice interface for managing controllers. It would be good
to add similar functionality to Mame for tvos, although I wouldn't know how
to do that.

**Comment by JonnyBoy333** [comment]
Anybody find a way to get two controllers to register yet? I figured out that holding right and down while pressing start starts players 3 and 4. Like ian said, though, all players are controlled by the one controller. On the bright side I found if you do this in the Simpsons and all 4 players are on top of each other they form a ball and you roll around as some sort of invincible death ball for about 30 seconds..

**Comment by romadu** [comment]
LesBird updated mame4ios. It now supports multiple controllers. https://github.com/yoshisuga/MAME4iOS
On my iPhone both nimbus show as player 1 (the second player replaces the first). Pushing menu+b adds coins. Pressing menu adds a second player into the game. I could use 2 controllers and 2 separate people for street fighter 2 and 1943, and AirPlay to AppleTV

**Comment by JonnyBoy333** [comment]
Cool, I saw they were talking about creating a tvOS port for that version but it doesn't look like it's been worked on in a while. Hopefully someone gets back at it.

**Comment by romadu** [comment]
Yes, it seems they didn't have time yet. At least AirPlay from IOS works
with multiple controllers for now (and perhaps the controller code can be
ported to tvos version)

**Comment by JonnyBoy333** [comment]
And just like that it looks like the tvOS port is under development. This build has native support for mFi controllers and uses a more recent mame emulator version (less bugs). If you think this is cool go over and give him a thumbs up to show your support.
https://github.com/yoshisuga/MAME4iOS/issues/1

**Comment by romadu** [comment]
Good news, but a set-back
"Oops I take that back - I wasn't able to get the MAME library to compile
for arm64. There are some issues relating to casting to 32-bit ints, and I
wasn't able to quickly fix it. If anyone can checkout the tvos-arm64 branch
and try to get the mame library to compile, it'd be rather trivial to get
this working under tvOS."
